### ISSUE TITLE
E2guardian crashes on centos 7

### ISSUE THREAD

**Issue by fcruzcoelho** [e2guardian/e2guardian#188]
Good afternoon,
I am going through a problem of authentication of users outside the domain that use S.O Windows, I use NTLM authentication, when I open the browser it does not request authentication and tries to navigate with the local user of the machine which has no access configured to navigate.

I made tests with S.O Linux outside of the domain and when I open the browser it requests user/password usually validating the access filter in filtergroupslist according to group.

To validate my NTLM authentication configuration I took E2guardian from the front and set my browser to go straight through squid, where the user/password request normally appears.

Could you inform me if I have to perform any specific configuration in E2guardian so that machines with S.O windows outside the domain ask for user/password to authenticate. Or is it a bug that has already been fixed or has been fixed in the new version 3.5

thank you

**Comment by fredbcode** [comment]
Which version ?

**Comment by fcruzcoelho** [comment]
my verison is e2guardian 3.4.0.3

thank you

**Comment by fredbcode** [comment]
Ok, there are many changes in 3.5.1 (about NTLM) please make a try

**Comment by fcruzcoelho** [comment]
Ok, I will perform the build of version 3.5.1, but I still need to relatify the settings related to file descriptors ??

I use CentOS7 would the RPM package for installation? In case I do not compile myself.

Thank you

**Comment by fcruzcoelho** [comment]
@fredbcode 

I upgraded my e2g to version 3.5.1,

But I can not authenticate anymore I'm getting errors in / var / log / messages follow print.
![snapshot120](https://cloud.githubusercontent.com/assets/13903548/24057112/fc25e210-0b24-11e7-849c-a98dbbe66c44.png)
Do you know if I have to configure any option?

Thank you

**Comment by fredbcode** [comment]
How, compilation, package ?

**Comment by fcruzcoelho** [comment]
I made the installation via package like the previous one, I generated the package by the spec file that is in the google group, I just changed the versions.

I've just made an appointment from scratch and apparently re-authenticated and navigated.

I'm doing some more testing as soon as I finish post here.

Thank you

**Comment by fcruzcoelho** [comment]
@fredbcode 
With the default configuration it allows to navigate after enabling the filtergroupslist, returns to not authenticate and thus I can not navigate. And the e2g service is running normally, but the errors continue

tail -f /var/log/messages
Mar 17 15:42:18 btu-squid-hml kernel: e2guardian[16863]: segfault at e7a87d0 ip 0000000000424413 sp 00007ffd75a9e020 error 4 in e2guardian[400000+e7000]
Mar 17 15:42:18 btu-squid-hml kernel: e2guardian[16865]: segfault at e7a87d0 ip 0000000000424413 sp 00007ffd75a9e020 error 4 in e2guardian[400000+e7000]
Mar 17 15:42:18 btu-squid-hml kernel: e2guardian[16861]: segfault at e7a87d0 ip 0000000000424413 sp 00007ffd75a9e020 error 4 in e2guardian[400000+e7000]
Mar 17 15:42:18 btu-squid-hml kernel: e2guardian[16866]: segfault at e7a87d0 ip 0000000000424413 sp 00007ffd75a9e020 error 4 in e2guardian[400000+e7000]
Do you know if I have to configure any option?

Thank you

**Comment by fredbcode** [comment]
No there is something wrong, which os ?
I guess the problem is with the binary not adapted to your system, but why ?
My package works with Jessie 64 bits and I guess Ubuntu 16.04 64 bits (not tested by me)
Please can you be more explicit, explain your installation step by step ?

Le 17 mars 2017 19:45:00 GMT+01:00, fcruzcoelho <notifications@github.com> a écrit :
>@fredbcode 
>With the default configuration it allows to navigate after enabling the
>filtergroupslist, returns to not authenticate and thus I can not
>navigate. And the e2g service is running normally, but the errors
>continue
>
>tail -f /var/log/messages
>Mar 17 15:42:18 btu-squid-hml kernel: e2guardian[16863]: segfault at
>e7a87d0 ip 0000000000424413 sp 00007ffd75a9e020 error 4 in
>e2guardian[400000+e7000]
>Mar 17 15:42:18 btu-squid-hml kernel: e2guardian[16865]: segfault at
>e7a87d0 ip 0000000000424413 sp 00007ffd75a9e020 error 4 in
>e2guardian[400000+e7000]
>Mar 17 15:42:18 btu-squid-hml kernel: e2guardian[16861]: segfault at
>e7a87d0 ip 0000000000424413 sp 00007ffd75a9e020 error 4 in
>e2guardian[400000+e7000]
>Mar 17 15:42:18 btu-squid-hml kernel: e2guardian[16866]: segfault at
>e7a87d0 ip 0000000000424413 sp 00007ffd75a9e020 error 4 in
>e2guardian[400000+e7000]
>Do you know if I have to configure any option?
>
>Thank you
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/e2guardian/e2guardian/issues/188#issuecomment-287439715

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by spikedrba** [comment]
fwiw Ubuntu 16.04 definitely works, I run on that with hundreds of clients behind it and no problems, certainly no segfaults.

**Comment by fcruzcoelho** [comment]
@fredbcode 
I am using CentOS7.
I am doing the compiled installation on a server that is already running version 3.4.0.3.

Installation steps

I'm doing the build according to the wiki of voices, but I'm generating an RPM package from a spec file as documented in the google group.

Follow Link below:

With this file I have successfully created the RPM package for version 3.4.0.3.

Changing only the parameters to version 3.5.1 can also successfully create the RPM 3.5.1 package from that rpm I successfully update but I get the message from segfault until after restart.

I'm doing other tests, but still unsuccessful.

Thank you

**Comment by fcruzcoelho** [comment]
@fredbcode 

In this new version still need to change the config in the files below?
/usr/include/bits/typesizes.h
And
/usr/include/linux/posix_types.h

Thank you

**Comment by fredbcode** [comment]
Yes if --filedescriptors beyond 1024

**Comment by fcruzcoelho** [comment]
@fredbcode 

I made the compilation of version 3.5.1 as are the steps in the wiki, but I still can not get through e2g. Follows errors below.
![proxy](https://cloud.githubusercontent.com/assets/13903548/24159535/0d550e14-0e3e-11e7-8ce9-3801d2d008f9.jpeg)
Do you know what it can be?

Thank you

**Comment by fredbcode** [comment]
Unfortunately this is the same problem, did you compile on the same machine ?

**Comment by fredbcode** [comment]
OS and libraries must be exactly the same. It would be bad memory, try running memtest86+

**Comment by fcruzcoelho** [comment]
Yes I am compiling on the same machine, strange that it updates normal. But it does not let you navigate.

I am setting up a server with Debian to test the .deb package and see if it with my problem of browsing with machines outside the domain is solved.

Thank you

**Comment by fredbcode** [comment]
Can you make a try without ntlm ?

**Comment by fcruzcoelho** [comment]
@fredbcode 
In squid or E2guardian?

Thank you

**Comment by fredbcode** [comment]
Both please, just to known if there is a relation

**Comment by fcruzcoelho** [comment]
@fredbcode 

I made a new installation zeroed using the Debian8 64bit OS. I set the squid3 normally and without the e2g 3.5.1 on the front navigation works perfectly with either NTLM or NTLM.

I downloaded the package e2guardian_3.5.1_jessie_amd64.deb, and I installed it according to wiki, with the default settings without NTLM the navigation works without problems, after enabling the NTLM and/or Basic authentication plugin I can not navigate anymore.

Looking at the logs in / var / log / messages I get the same error as in CentOS7.

Follow Print:
![snapshot121](https://cloud.githubusercontent.com/assets/13903548/24206552/51509f88-0efd-11e7-8b90-0f62a8e6efca.png)
It looks like it's trying to get out to IPv6 I do not know. In version 3.4.0.3 the operation is perfect as in Debian and/or CentOS7.

But since I'm having NTLM issues and version 3.5.1 has several NTLM related runs I need to upgrade to tests, but I can not get it to work.

For discharging I also did tests on Ubuntu 16.04.2 and I have the same problem all with an installation of S.O from scratch.

Do not I need to do some configuration in e2guardian.conf other than version 3.4.0.3?

I have already done several tests and all for the same problem.

Thank you

**Comment by fredbcode** [comment]
Very weird, I'm using the same configuration without any problem 
Can you copy and past your configuration ?
Do you think I can connect to your machine ?
How many maxchildren ?  under 1024 ?

**Comment by fredbcode** [comment]
> It looks like it's trying to get out to IPv6

Why do you say that ? You can easily disabled ipv6

**Comment by fredbcode** [comment]
Same problem with "Basic authentication" alone ?

**Comment by fcruzcoelho** [comment]
IPv6 It was just an option I do not know where the error is !! I'll disable.
Yes in basic too.

I am at lunchtime can be after 2:30 pm barsil hours?
Thank you

**Comment by fredbcode** [comment]
Can made a test machine with a ssh access ? If yes you can send the credential to numsys - no spam - @ - no spam - .free.fr

**Comment by fredbcode** [comment]
I'm thinking about something, are you using the same hardware (or VM host) ?
If yes can you make a try with an another physical machine ?

**Comment by fredbcode** [comment]
Of course the dependencies are installed ? 
Depends: adduser, perl, libbz2-1.0, libc6 (>= 2.4), libgcc1 (>= 1:4.1.1), libpcre3 (>= 8.10), libstdc++6 (>= 4.6), libtommath0, zlib1g (>= 1:1.1.4)
Suggests: squid, clamav (>= 0.80), clamav-freshclam

I'm testing right now with a new Debian 8 64 bits and basic auth, no problem

**Comment by fcruzcoelho** [comment]
Can I release a teamviewer access?

I'm using a VM in Hyper -V I do not have a physical machine to install one and do tests.

On the dependencies I believe that I have all installed yes until when I installed e2g via dpkg -i did not complain about any.

But could you send me all the dependencies possible for me to verify?

Thank you

**Comment by fredbcode** [comment]
Yes you have in my previous message
 VM in Hyper -V Ok I have an idea can try to recompile with this options

./autogen.sh
 '--prefix=/usr' '--enable-clamd=yes' '--with-proxyuser=e2guardian' '--with-proxygroup=e2guardian' '--sysconfdir=/etc' '--localstatedir=/var' '--enable-icap=yes' '--enable-commandline=yes' '--enable-email=yes' '--enable-ntlm=yes' '--enable-trickledm=yes' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/
info' '--enable-pcre=yes' '--with-filedescriptors=8192' '--enable-sslmitm=yes'
make

**Comment by fredbcode** [comment]
And you switch the binary in /usr/sbin/e2guardian by the new one src/e2guardian

**Comment by fcruzcoelho** [comment]
Okay, I'm restoring a snapshot of my VM to perform the build as per its orientation.

Thank you

**Comment by fredbcode** [comment]
First test change with-filedescriptors=8192 by with-filedescriptors=1000 (just for testing)
And try with maxchildren=900 in e2guardian.conf

**Comment by fcruzcoelho** [comment]
S.O Debian Already with ulimit -n to 65536 by default

root @ btu-e2g: ~ # ulimit -n
65536

is there any problem?

Tks

**Comment by fredbcode** [comment]
No

Le 22 mars 2017 18:59:01 GMT+01:00, fcruzcoelho <notifications@github.com> a écrit :
>S.O Debian Already with ulimit -n to 65536 by default
>
>root @ btu-e2g: ~ # ulimit -n
>65536
>
>is there any problem?
>
>Tks
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/e2guardian/e2guardian/issues/188#issuecomment-288486046

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by fcruzcoelho** [comment]
Hi, 

I have already installed all the lib error that has reported any more?

./configure: line 6848: PKG_PROG_PKG_CONFIG: command not found
checking for PCRE support... yes
./configure: line 6875: syntax error near unexpected token `PCRE,libpcre'
./configure: line 6875: `PKG_CHECK_MODULES(PCRE,libpcre >= 6.0)'

tks

**Comment by fredbcode** [comment]
Hum, try with pcre=no

Le 22 mars 2017 19:19:26 GMT+01:00, fcruzcoelho <notifications@github.com> a écrit :
>Hi, 
>
>I have already installed all the lib error that has reported any more?
>
>./configure: line 6848: PKG_PROG_PKG_CONFIG: command not found
>checking for PCRE support... yes
>./configure: line 6875: syntax error near unexpected token
>`PCRE,libpcre'
>./configure: line 6875: `PKG_CHECK_MODULES(PCRE,libpcre >= 6.0)'
>
>tks
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/e2guardian/e2guardian/issues/188#issuecomment-288492436

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by fredbcode** [comment]
Please, there is all files needed in INSTALL file on github
See debian part

Le 22 mars 2017 19:19:26 GMT+01:00, fcruzcoelho <notifications@github.com> a écrit :
>Hi, 
>
>I have already installed all the lib error that has reported any more?
>
>./configure: line 6848: PKG_PROG_PKG_CONFIG: command not found
>checking for PCRE support... yes
>./configure: line 6875: syntax error near unexpected token
>`PCRE,libpcre'
>./configure: line 6875: `PKG_CHECK_MODULES(PCRE,libpcre >= 6.0)'
>
>tks
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/e2guardian/e2guardian/issues/188#issuecomment-288492436

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by fcruzcoelho** [comment]
I looked and installed all dependencies that are in INSTALL NO GIT, but still giving me the same error, even with pcre=no.

./configure: line 6848: PKG_PROG_PKG_CONFIG: command not found
checking for PCRE support... no
./configure: line 6875: syntax error near unexpected token `PCRE,libpcre'
./configure: line 6875: `PKG_CHECK_MODULES(PCRE,libpcre >= 6.0)'
root@btu-e2g:/usr/local/src/e2guardian-3.5.1# 

Searching!!

**Comment by fcruzcoelho** [comment]
Solved.
run command autoreconf -vfi.
setting to test.

Tks

**Comment by fredbcode** [comment]
I Wonder why it is needed for you ...
E2guardian runs ?

----- fcruzcoelho <notifications@github.com> a écrit :

**Comment by fcruzcoelho** [comment]
It does all part of locks / Release on the Internet.

I did not understand the question

**Comment by fredbcode** [comment]
I mean it works now ? E2guardian ?

**Comment by fcruzcoelho** [comment]
No, giving permission errors, I suppose.
root@btu-e2g:/etc/e2guardian# systemctl status e2guardian.service 
● e2guardian.service - E2guardian Web filtering
   Loaded: loaded (/etc/systemd/system/e2guardian.service; enabled)
   Active: failed (Result: signal) since Wed 2017-03-22 17:06:43 -03; 1min 56s ago
  Process: 5563 ExecStart=/usr/sbin/e2guardian (code=killed, signal=SEGV)
 Main PID: 982 (code=exited, status=0/SUCCESS)

Mar 22 17:06:43 btu-e2g e2guardian[5563]: Required Listname authexceptionsitelist is not defined
Mar 22 17:06:43 btu-e2g e2guardian[5563]: Error opening filter group config: /etc/e2guardian/e2guardianf2.conf
Mar 22 17:06:43 btu-e2g e2guardian[5563]: Error reading filter group conf file(s).
Mar 22 17:06:43 btu-e2g e2guardian[5563]: Error parsing the e2guardian.conf file or other e2guardian configuration files
Mar 22 17:06:43 btu-e2g e2guardian[5563]: Error opening filter group config: /etc/e2guardian/e2guardianf2.conf
Mar 22 17:06:43 btu-e2g e2guardian[5563]: Error reading filter group conf file(s).
Mar 22 17:06:43 btu-e2g e2guardian[5563]: Error parsing the e2guardian.conf file or other e2guardian configuration files
Mar 22 17:06:43 btu-e2g systemd[1]: e2guardian.service: control process exited, code=killed status=11
Mar 22 17:06:43 btu-e2g systemd[1]: Failed to start E2guardian Web filtering.
Mar 22 17:06:43 btu-e2g systemd[1]: Unit e2guardian.service entered failed state.
root@btu-e2g:/etc/e2guardian#

**Comment by fredbcode** [comment]
Yes the owner must be e2guardian on /etc/e2guardian and /var/log/e2guardian

**Comment by fredbcode** [comment]
On all files recursively

**Comment by fcruzcoelho** [comment]
I already executed it. like this. below
root@btu-e2g:~# chown e2guardian:e2guardian /etc/e2guardian/ -R
root@btu-e2g:~# chown e2guardian:e2guardian /var/log/e2guardian/ -R

**Comment by fcruzcoelho** [comment]
When I disable authentication ntlm starts successfully.
I'm going to change environment for testing, it's not possible.

**Comment by fredbcode** [comment]
Install the package and just change the binary

**Comment by fcruzcoelho** [comment]
I already did the e2guardian binary change and nothing.

**Comment by fredbcode** [comment]
I don't understand

**Comment by fcruzcoelho** [comment]
Neither is it very strange, since version 3.4.0.3 is in the same environment and working.

I've been with this problem all day and nothing. It's confusing.

Thank you for your help

**Comment by fcruzcoelho** [comment]
@fredbcode 
I have refitted all configuration and believe that I discovered the problem.
I configured 3 groups as below.
Defaults - (0 = banned)
Advanced (1 = filtered)
Intermediary (1 = filtered)

And gave the appropriate permissions to the users in filtergrouplist for each group

With these settings the error continues, in the group defaults that is e2guardianf1.conf enabled the filter (1 = filtered) immediately the navigation worked with the ntlm authentication normally but when I return the group defaults to (0 = banned) all navigation and authentication to in hour.

In version 3.4.0.3 I have just this scenario and it works perfectly, the user who is not part of any group falls in the defaults denying any navigation.

Do you use this scenario?

Thank you

**Comment by fredbcode** [comment]
Yes right I'm testing right now there is a bug !

So you can "fix" this with adding ** in bannedsitelist, in this case all site will be banned

**Comment by fcruzcoelho** [comment]
@fredbcode 
Good Morning,
In my production environment I used just this way, but I wanted to use the groupemode = 0 mode in the default group, for an internal control.

I will now configure my homologation environment equal to the production and let the bug be corrected.

Thanks for the help even here.

**Comment by fredbcode** [comment]
Ok we will remove groupmode soon, this is a confusing feature (for code and admin)

Thanks for your help 


>

**Comment by fcruzcoelho** [comment]
@fredbcode 
I'm migrating my environment to Debian and using the .deb package, but I'm getting the message from below.
![snapshot122](https://cloud.githubusercontent.com/assets/13903548/24375604/5130e36c-130f-11e7-9728-c8c2c8ece8b4.png)


I have already made the change in posix_types.he typesizes.h files for a larger number of FD, but the same does not comply with the configuration notem that in the debian file typesizes.h, it is in another directory "/ usr / include / x86_64-linux- Gnu / bits # "other than CentOS7 will that be the problem?

It has some more configuration to work with a greater number of FD in S.O Debian8

Thank you

**Comment by fcruzcoelho** [comment]
@fredbcode 
Sorted out,
Include the line "Limit NOFILE = 8192" in /etc/systemd/system/e2guardian.service then restart and function normally.
If I want to include it in your package.
Thank you

**Comment by fredbcode** [comment]
Yes just forget in package, fixed now

Thanks

----- Mail original -----

**Comment by fcruzcoelho** [comment]
@fredbcode 
I made all the settings and the operation is perfect, however in S.O windows outside the domain does not appear the popup box to enter user and password, already in S.O Linux appears tranguilo.

Can you tell me if there is a bug like this:
I tested it on every possible browser.
Chrome
IE
Firefox

Thank you

**Comment by fredbcode** [comment]
Sorry, I am not an expert about NTLM
Do you have the same issue with e2guardian disabled ? If yes you can try to ask your question to the squid mailing list ?

**Comment by fcruzcoelho** [comment]
@fredbcode 

With e2guardian disabled the popup box appears normally, and authenticates successfully.
I believe the e2g filter does not support some settings for groups. I'm looking at the best way to perform filtergroup settings.

Thank you

**Comment by fcruzcoelho** [comment]
@fredbcode 
I made several tests with the filtergrouplist option and could not reach a definitive configuration, to enable popup on Windows machines outside the domain.
I have 5 groups in the following sequence. All with the option of groupmode = 1

Restricted     | (e2guardianf1.conf)
Advanced     | (e2guardianf2.conf)
Intermediary | (e2guardianf3.conf)
Mkt               | (e2guardianf4.conf)
Basics          | (e2guardianf4.conf)

In the restricted group this is fixed in bannedsitelist like ** to block all the sites and to release only what is in exceptionsitelist (whitelist)

When the computers are in the domain the filter is applied successfully and works very well, but when a machine with S.O. Windows is out of the domain Popup screen does not appear failing to authenticate and thus use the filter applied to that user.

In my tests I noticed that the PopUP screen appears only when I try to access a website that is added in whitelist (exceptionsitelist)

Another action you notice is that it does the first filter in the first group that is in e2guardianf1.conf.
I believe that for other scenarios the popup works because the default group is not blocking everything, which I can not leave in my environment. There are many users who do not need internet access. And yes just what is in whitelist (exceptionsitelist)

My question is, this would be a bug/improvement or the behavior is that same.

I believe that the filter of the default group should be applied as soon as the user authenticates and validates the accesses. (Suggestion only)

Thank you

**Comment by fredbcode** [comment]
> In the restricted group this is fixed in bannedsitelist like ** to 
> block all the sites and to release only what is in exceptionsitelist 
> (whitelist)
>
> When the computers are in the domain the filter is applied 
> successfully and works very well, but when a machine with S.O. Windows 
> is out of the domain Popup screen does not appear failing to 
> authenticate and thus use the filter applied to that user.
>
> In my tests I noticed that the PopUP screen appears only when I try to 
> access a website that is added in whitelist (exceptionsitelist)
>
> Another action you notice is that it does the first filter in the 
> first group that is in e2guardianf1.conf.
> I believe that for other scenarios the popup works because the default 
> group is not blocking everything, which I can not leave in my 
> environment. There are many users who do not need internet access. And 
> yes just what is in whitelist (exceptionsitelist)
>

In NTLM with domain there is no Pop-up at all, I don't know about pop-up 
with NTLM (outside the domain) and how it can be works
Are you sure this pop-up is not more related basic auth  ?

Please, can you try with a plugin like 
https://addons.mozilla.org/fr/firefox/addon/live-http-headers/
And post the result ?

A test with pop-up of course

**Comment by fcruzcoelho** [comment]
You are right, with NTLM does not need popup, but I have machines that are not part of the domain and the same can not authenticate when trying to access any page on the internet, ie the navigation tries to leave the local user, which is not part Of no group filters.

My question is because on machines outside a domain, does the popup not appear when the user tries to access any website? In my case the popup only appears when you access a site that is part of the whitelist.

Can you understand my doubt?

Sorry if I'm being confused.

Thank you

**Comment by fcruzcoelho** [comment]
@fredbcode 

I'm having a lot of problems on government websites "gov.br" several at some point requests authentication again and does not open the page, with that I believe I'm having an overload on my server.

The problem happens in the majority when they are sending some file or authenticating in some panel.

Follow Print.
![hd-358995](https://cloud.githubusercontent.com/assets/13903548/24724618/22f245e4-1a22-11e7-8de9-b89a1407d10b.JPG)


Have you had this problem? 
This happens when you have some service running java.
Thank you

**Comment by fredbcode** [comment]
If I understand right, you have a whitelist (so no pop-up for some websites) right ?
And I guess gov.br is a part of them

If yes:

There is certainly some part of this website outside the domain allowed, you can confirm this with squid log there is certainly a 407 DENIED (mean authentication wrong/nedeed)with a completely different domains 

How you can fix this

1 - Add a basic authentication, it can be a generic account for all
Or 
2 - Open the domains one by one
Or
3 - Disabled authentication for the user source IP address (in squid)

Fred 



 

----- Mail original -----

**Comment by fredbcode** [comment]
> different domains
> 

I mean web domains, nothing related with NTLM

**Comment by fcruzcoelho** [comment]
@fredbcode 

"If I understand right, you have a whitelist (so no pop-up for some websites) right ?
And I guess gov.br is a part of them"
---------------------------------------------
That's right, the popup only appears for sites that are on the white list, is this normal, or can it be a bug? I think the popup should appear for any site, from the default group, which is the first filter to be read by e2g.

The gov.br sites are related to the 407 authentication error, I am verifying.

Thank you

**Comment by fredbcode** [comment]
I guess this is a problem with squid not e2, the pop-up is generated by squid himself 
Can you make a try without e2, squid only ?

**Comment by fcruzcoelho** [comment]
I already did this test, when this is just squid, the popup appears normally, for any site.

Thank you

**Comment by fcruzcoelho** [comment]
@fredbcode 
In my tests all access works perfect without the e2g in front, I have no authentication errors (407), nor popup appearing for some sites, randomly.

In my old environment I used SquidGuard for content blocking, and it also did not have any such problems.

The e2g works fine, but I'm having a lot of NTLM-related issues I did not have in the old environment.

I am redoing my environment with SquidGuard to perform new tests.

Thanks for the help even here.

**Comment by fredbcode** [comment]
Di you have this issue with previous version 3.4, 3.3, etc ?

**Comment by fcruzcoelho** [comment]
I have this problem with version 3.4.0.3 and 3.5.1, in my production environment this is with version 3.4.3 and the approval is 3.5.1, where I have the same problems.

Everyone works when I shoot the front e2g, and I only leave the squid.

Does e2g use another cache service? Without being squid?

Thank you

**Comment by philipianpearce** [comment]
Just to clarify a few things: 

    1. The pop-up login box is nothing to do with e2g or squid. It is thrown up by the browser only when it does not have access to the username/password required when the proxy server sends a 407 response. There is always a http 407 response the first time a browser tries to use an authenticated proxy. It is just that when browser is logged in to the local windows domain it is able to supply these credentials without challenging the user for them. 
    2. e2g is unable to sniff usernames from NTLMv2 or Netbios logins so NTLM type 1 requests must be enabled on every browser for it to work, as it is disabled by default. 
    3. e2g can also use basic or digest proxy auth. 
    4. Is squid configured to fail back to basic auth when NTLM fails? If so, it could be that the Win SO units are refusing to supply NTLM credentials and squid is then asking for basic. If this is the case, you will need to enable the e2g 'basic' auth-plugin as well as the NTLM one in order for e2g to be able log usernames. 
    5. In any case you need to completely understand what is going on at the browser end to work out what the difference is between going direct to squid and going via e2g is and what proxy authentication protocols are actually being used. 
    6. I have never used any pass through auth with e2g, but as far as I am aware, it has always been a requirement that in order to log-in with pass through auth you have to visit a url allowed by filtergroup 1 for the first request from the browser (and in all cases, with or without user pop-up). This logic is likely to be changed in the next version (i.e. post 4.1) as all of the filtering flow logic will be undergoing a major revision. However, to change it in the current versions would in my view risk de-stabilizing the code, and may have unexpected side-effects on implementations which expect the current behavior. 
    7. 
        * As a work round in your set up this allowed page could be just a 'login' page or conditions of use page on one of your sites and it may be possible to use 
	urlredirectregexplist to globally redirect all other requests to the allowed one. This should work at least for http. 

Regards 

Philip 

----- Original Message -----

**Comment by fcruzcoelho** [comment]
@philipianpearce 
First of all, thank you for the explanation.

In my scenario the authentication part with windows machines outside the local domain, was resolved according to your comment, where users access a URL that is part of a whitelist and thus enabling the popup to authenticate successfully.

About problem 407 I solved with some ACLs not to request authentication. In my case the vast majority are government websites.

Follow:
acl governo dstdomain .gov.br
always_direct allow governo
cache deny governo

acl semcache url_regex -i "/blacklists/squid/acls/semcache" \?
no_cache deny semcache

How good that in version 4.1 you are reviewing the filters issue.

Thank you
